src/gz/bufread.rs

```diff
@@ -449,9 +453,13 @@ impl<R: BufRead> MultiGzDecoder<R> {
 }
 
 impl<R> MultiGzDecoder<R> {
-    /// Returns the current header associated with this stream, if it's valid
-    pub fn header(&self) -> Option<&GzHeader> {
-        self.0.header()
+    /// Returns the headers processed so far
+    pub fn headers(&self) -> Vec<&GzHeader> {
```

Comment:
This could be allocation-free by returning `impl Iterator<Item = &GzHeader>` with this implementation:

```Rust
        self.0.headers.iter().chain(self.0.header())
```